lib/tasks/convert_subject_fields_to_subject_array.rake

```diff
@@ -0,0 +1,39 @@
+namespace :data do
+  desc 'Convert subject fields to string array for vacancies without the new subject field'
+  namespace :convert_subject_fields do
+    task vacancies: :environment do
+      updated_count = 0
+      should_be_updated_count = Vacancy.where("subjects is NULL or subjects = '{}'").count
+
+      Rails.logger.info(
+        "Conversion of subject fields to string array has been started for #{should_be_updated_count} vacancies"
+      )
+      Rollbar.log(
+        :info,
+        "Conversion of subject fields to string array has been started for #{should_be_updated_count} vacancies"
+      )
+
+      Vacancy.where("subjects is NULL or subjects = '{}'").in_batches(of: 100).each_record do |vacancy|
+        subject_name = vacancy.subject.present? ? vacancy.subject.name : nil
+        first_subject_name = vacancy.first_supporting_subject.present? ? vacancy.first_supporting_subject.name : nil
```

Comment:
Here you can use the `presence` method, is more concise:
https://andycroll.com/ruby/use-the-presence-method

Reply:
Would this return the name or the object?

Reply:
`vacancy.subject.name.presence`